runtime/src/main/java/io/micronaut/jackson/deserialize/StringTrimModule.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.jackson.deserialize;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.StringDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.util.StringUtils;
+
+import javax.inject.Singleton;
+import java.io.IOException;
+
+@Singleton
+@Requires(property = "jackson.trim-strings", value = StringUtils.TRUE)
+public class StringTrimModule extends SimpleModule {
+
+    public StringTrimModule() {
+        addDeserializer(String.class, new StringDeserializer() {
```

Comment:
This should probably be done within the `ObjectMapperFactory` and then you can check for `jacksonConfiguration.getTrimStrings()`